### PR TITLE
feat:[AH-268]: schema changes for CI docker push step

### PR DESCRIPTION
**Pull request by sourabh-awashti**

### CODE REVIEW

File: v0/pipeline/steps/ci/docker-step-info.yaml [hunk 1]
```diff
@@ -2,10 +2,16 @@ title: DockerStepInfo
...
-  - connectorRef
-  - repo
-  - tags
+  oneOf:
+   - required:
+      - connectorRef
+      - repo
+      - tags
+   - required:
+       - registryRef
```
Comment (arvind-choudhary-h) [inline v0/pipeline/steps/ci/docker-step-info.yaml:11]: The indentation looks different here.

File: v0/pipeline/steps/ci/docker-step-info.yaml [hunk 2]
```diff
@@ -2,10 +2,16 @@ title: DockerStepInfo
...
+       - useHarnessRegistry
+       - repo
+       - tags
```
Comment (arvind-choudhary-h) [inline v0/pipeline/steps/ci/docker-step-info.yaml:14]: today, are repo and tags required fields?

Comment (shivagowda) [reply v0/pipeline/steps/ci/docker-step-info.yaml:14]: yes, they are required fields

**Review by shivagowda — DISMISSED** [review]

File: v0/pipeline/steps/ci/docker-step-info.yaml (hunk 1 above)
Comment (sourabh-awashti) [reply v0/pipeline/steps/ci/docker-step-info.yaml]: done

File: v0/pipeline/steps/ci/docker-step-info.yaml (hunk 2 above)
Comment (sourabh-awashti) [reply v0/pipeline/steps/ci/docker-step-info.yaml]: yes, they both are required

File: v0/pipeline/steps/ci/docker-step-info.yaml [hunk 3]
```diff
@@ -2,10 +2,16 @@ title: DockerStepInfo
...
+  oneOf:
+   - required:
+      - connectorRef
+      - repo
+      - tags
+   - required:
+      - registryRef
+      - useHarnessRegistry
+      - repo
+      - tags
```
Comment (tmacari-harness) [inline v0/pipeline/steps/ci/docker-step-info.yaml:14]: so user can configure one of: 
```
      - connectorRef
      - repo
      - tags
      ```
      or
      ```
      - registryRef
      - useHarnessRegistry
      - repo
      - tags
      ```
     what if in second case user will set **useHarnessRegistry** false?
     we can end with an inconsistent config.
     
     Why do we need **useHarnessRegistry**  at all?
    we have 
    ```
      - repo
      - tags
      ```
      which are common fields, and we have **connectorRef** and **registryRef**, which can go under **oneOf**.
      @arvind-choudhary-h , Harsh Jain, what do you think?

Comment (arvind-choudhary-h) [reply v0/pipeline/steps/ci/docker-step-info.yaml:14]: to me, it's makes sense to not use `useHarnessRegistry`

Comment (sourabh-awashti) [reply v0/pipeline/steps/ci/docker-step-info.yaml:13]: done

**Review by harshjain12 — DISMISSED** [review]

**Review by tmacari-harness — DISMISSED** [review]

**Review by tmacari-harness — APPROVED** [review]

**Review by harshjain12 — APPROVED** [review]

**Pull request merged**